src/main/java/frc/robot/commands/vision/CalculateStdDevs.java

```diff
@@ -0,0 +1,139 @@
+package frc.robot.commands.vision;
+
+import java.util.ArrayList;
+
+import org.photonvision.EstimatedRobotPose;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.LogManager;
+import frc.robot.util.Vision;
+
+/**
+ * Calculates standard deviations for vision
+ */
+public class CalculateStdDevs extends CommandBase {
+  private final ShuffleboardTab m_tab;
+  private final Drivetrain m_drive;
+  private final Vision m_vision;
+  private ArrayList<Pose2d> m_poses;
+  private int m_arrayLength;
+  private Timer m_endTimer;
+
+  /**
+   * Constructor for CalculateStdDevs
+   * @param time How long to run the command, also how many poses to use
+   * @param shuffleboardTab The shuffleboard tab to output to
+   * @param drive The drivetrain
+   * @param vision The vision
+   */
+  public CalculateStdDevs(int time, ShuffleboardTab shuffleboardTab, Drivetrain drive, Vision vision) {
+    m_tab = shuffleboardTab;
+    m_drive = drive;
+    m_vision = vision;
+    m_arrayLength = time;
+  }
+
+  /**
+   * Resets the pose array
+   */
+  @Override
+  public void initialize() {
+    m_poses = new ArrayList<Pose2d>();
+  }
+
+  /**
+   * Adds a pose to the array
+   */
+  @Override
+  public void execute() {
+    Pose2d pose = m_vision.getPose2d(m_drive.getPose());
+    // If the pose exists, add it to the first open spot in the array
+    if (pose != null) {
+      m_endTimer.stop();
+      m_endTimer.reset();
+      m_poses.add(pose);
+      System.out.printf("%.1f%% done", ((double)m_poses.size())/m_arrayLength * 100);
+    } else {
+      m_endTimer.start();
+      // If 5 seconds have passed since it saw an April tag, stop the command
+      // Prevents it from running forever
+      if(m_endTimer.hasElapsed(5)){
+        cancel();
+      }
+    }
+  }
+
+  /**
+   * Calculates the standard deviation
+   */
+  @Override
+  public void end(boolean interrupted) {
+    // If the array is empty, don't try to calculate std devs
+    if(m_poses.size() == 0){
+      System.out.println("There are no poses in the array\nTry again where the robot can see an April tag.");
+      return;
+    }
+    
+    // Calculate std devs
+    double meanX = 0;
+    double meanY = 0;
+    double meanRot = 0;
+    for (int i = 0; i < m_poses.size(); i++) {
+      meanX += m_poses.get(i).getX();
+      meanY += m_poses.get(i).getY();
+      meanRot += m_poses.get(i).getRotation().getRadians();
```

Comment:
It is wrong to compute mean rotation by averaging angle. Average of 0 and 360 degrees is 0 (or 360), but using this logic it will be 180. A simple way to do it correctly is to operate on Rotation objects - sum them up and divide, and then take the angle

Reply:
I would like to point out that this will still result is avrageing issues. Example, 5 messuements of 90 degrees added together is 90 degreess then diveded by 5 is 18 degrees wich is clearly not the mean. Additionaly this is only an issue if the pose outputs a values around the boarder of the continous. So if the output is -pi to pi a mean of 0 won't cause issues and if it is 0 to 2pi then a mean of pi won't cause issues.

Reply:
I would suggest adding documentation to state wich aprial tags could be used for the test.

Reply:
I mentioned on Slack to only use red April tags because of the averaging, because the angle won't be near -π or π.

Reply:
After Houston, I'll add more documentation, but there are other things to do until then.